lib/sles4sap_publiccloud.pm

```diff
@@ -496,65 +494,106 @@ sub cleanup_resource {
+        for my $site (keys %{$topology->{Site}}) {
+            die("check_takeover [ERROR] Missing 'srPoll' field in topology output of Site->$site") unless defined($topology->{Site}->{$site}->{srPoll});
+            # As 'vhost' value is not part of the 'Site' key we need to pair it with corect 'Host' key
+            for my $host (keys %{$topology->{Host}}) {
+                die("check_takeover [ERROR] Missing 'vhost' field in topology output of $host") unless defined($topology->{Host}->{$host}->{vhost});
+                $vhost = $topology->{Host}->{$host}->{vhost} if ($topology->{Host}->{$host}->{site} eq $site);
+                $sync_state = $topology->{Site}->{$site}->{srPoll} if ($topology->{Host}->{$host}->{site} eq $site);
+            }
+            record_info('Cluster Host', join("\n",
                     "vhost: $vhost compared with $hostname",
                     "sync_state: $sync_state compared with PRIM"));
             if ($vhost ne $hostname && $sync_state eq "PRIM") {
-                record_info("Takeover status:", "Takeover complete to node '$vhost");
+                record_info('Takeover status:', "Takeover complete to node '$vhost'");
                 last TAKEOVER_LOOP;
             }
         }
-        die("Test failed: takeover failed to complete.") if ($retry_count > 40);
+        die('check_takeover [ERROR] Test failed: takeover failed to complete.') if ($retry_count > 40);
         sleep 30;
     }
 
     return 1;
 }
 
 =head2 enable_replication
-    enable_replication([site_name => 'site_a']);
+    enable_replication([site_name => 'site_a', hana_inn => '10' ]);
 
-    Enables replication on fenced database. Database needs to be offline.
+    Re-enaobles replication on the previously fenced database node. Database must be offline.
+    Dies if the the given <site name> doesn't corespond with the SAPHanaSR topology local site name
+    Optionaly the hana_innstance_id cpould be provided for better relianility
 
 =over
 
-=item B<site_name> - site name of the site to register
+=item B<site_name> - local HANA site name of the site to be registered
+
+=item B<hana_inn> - optional HANA instance number to be registered
 
 =back
 =cut
 
 sub enable_replication {
     my ($self, %args) = @_;
-    croak("Argument <site_name> missing") unless $args{site_name};
+    croak('enable_replication [ERROR] Argument <site_name> missing') unless $args{site_name};
     my $hostname = $self->{my_instance}->{instance_id};
-    die("Database on the fenced node '$hostname' is not offline") if ($self->is_hana_database_online);
-    die("System replication '$hostname' is not offline") if ($self->is_primary_node_online);
+    my $remote_host;    # HANA site remote host to register
+    my $site = $args{site_name};    # HANA node name
+    my $sr_mode;    # HANA replication mode from SAPHanaSR-showAttr
+    my $op_mode;    # HANA SR operations mode from SAPHanaSR-showAttr
+    my $hana_inn;    # HANA instance number
```

Comment:
I see this line is here to have all local variable definitions at the beginning of the "scope" but maybe I'd move both this line and  line 560 both near to 578.

All in all how is working? User can provide a specific value for `hana_inn` from the external and if not a value is auto-calculated internally. Is it like this? Why we need to give this choice to user? How is calculated if not provided?